### PR TITLE
Split RandomMixin into prng and seed classes

### PR DESCRIPTION
**Pull request by theartful**
This has the artifact that if a parent class has the Seedable mixin, a derived class can't have RandomMixin (seed will be registered twice). It can be solved using SFINAE
```
RandomMixin : public Parent (if parent has Seedable)
RandomMixin : public Seedable<Parent> (if parent doesn't have Seedable)
```
but I'm not sure it's worth it, as it would complicate swig.

### CODE REVIEW

**Comment by karlnapf** [comment]
Is anyone against this, otherwise I'll merge it for use in xval

**Comment by iglesias** [comment]
+1.

**Comment by karlnapf** [comment]
I actually have one request here: Could you change CrossValidation to use the "seedable" mixin....since then we have a use-case straight away :)

**Comment by ghost** [comment]
@karlnapf done

**Comment by karlnapf** [comment]
Nice thanks!

File: src/shogun/evaluation/CrossValidation.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ namespace shogun
 	 * HP Laboratories.] for details on this subject.
 	 *
 	 */
-	class CCrossValidation : public RandomMixin<CMachineEvaluation>
+	class CCrossValidation : public Seedable<CMachineEvaluation>
```
Comment (karlnapf) [inline src/shogun/evaluation/CrossValidation.h:126]: nice!

**Pull request merged**